libraries/AP_Torqeedo/AP_Torqeedo.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_Torqeedo.h"
+
+#if TORQEEDO_ENABLED
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include <SRV_Channel/SRV_Channel.h>
+
+#define TORQEEDO_SERIAL_BAUD        19200   // communication is always at 19200
+#define TORQEEDO_PACKET_HEADER      0xAC    // communication packet header
+#define TORQEEDO_PACKET_FOOTER      0xAD    // communication packer footer
+
+extern const AP_HAL::HAL& hal;
+
+// parameters
+const AP_Param::GroupInfo AP_Torqeedo::var_info[] = {
+
+    // @Param: ENABLE
+    // @DisplayName: Torqeedo enable
+    // @Description: Torqeedo enable
+    // @Values: 0:Disabled, 1:Enabled
+    // @User: Standard
+    // @RebootRequired: True
+    AP_GROUPINFO_FLAGS("ENABLE", 1, AP_Torqeedo, _enable, 0, AP_PARAM_FLAG_ENABLE),
+
+    // @Param: ONOFF_PIN
+    // @DisplayName: Torqeedo ON/Off pin
+    // @Description: Pin number connected to Torqeedo's on/off pin. -1 to disable turning motor on/off from autopilot
+    // @Values: -1:Disabled,50:AUX1,51:AUX2,52:AUX3,53:AUX4,54:AUX5,55:AUX6
```

Comment:
I know this is probably swiped from somewhere else (if so, doesn't need fixing here but more genreally) - but this is horribly standard-Cube-carrier/Pixhawk1-specific

Reply:
Yes, it is swiped from AP_Camera and/or AP_LandingGear.  There are not many better examples of the list of available pins so I think I'll let this one go for now.

In a future PR I'm going to change the default so users use the Serial port's CTS pin to control On/Off by default which will make wiring easier.